workflow-ui/backstack-android/src/main/java/com/squareup/workflow1/ui/backstack/BackStackContainer.kt

```diff
@@ -136,23 +136,25 @@ public open class BackStackContainer @JvmOverloads constructor(
 
   override fun onRestoreInstanceState(state: Parcelable) {
     (state as? ViewStateCache.SavedState)
-        ?.let {
-          viewStateCache.restore(it.viewStateCache)
-          super.onRestoreInstanceState(state.superState)
-        }
-        ?: super.onRestoreInstanceState(state)
+      ?.let {
+        viewStateCache.restore(it.viewStateCache)
+        super.onRestoreInstanceState(state.superState)
+      }
+    // Some other class wrote state, but we're not allowed to skip
+    // the call to super. Make a no-op call.
+      ?: super.onRestoreInstanceState(super.onSaveInstanceState())
```

Comment:
The actual fix. Best to look at this ignoring whitespace.

Reply:
I guess this is actually super cheap because parcelables are lazy – sure looks weird though.

Reply:
Yeah. FWIW, I found this recipe in an ancient Stack Overflow, along with a lot of _really_ bad ones.